profiles/decorators.py

```diff
@@ -38,33 +56,18 @@ def wrapper(context, *args, **kwargs):
                     )
                 )
 
-            old_service = None
-            service_type = kwargs.get("service_type")
-            if service_type:
-                old_service = getattr(context, "service", None)
-                context.service = Service.objects.get(service_type=service_type)
+            if not context.service:
+                raise ServiceNotIdentifiedError(_("No service identified"))
```

Comment:
I would leave that as is. Now we only need one decorator in the resolvers `login_required`, `login_and_service_required` or `staff_required`. If the service required was its own decorator we would always need to add that decorator with one of the two `login_required` or `staff_required`.

And because the "beef" is only those two lines, I wouldn't like to make the decorators any more complex by separating those two lines to a helper decorator. The helper would need the context parameter and things could get hairy I think.

Reply:
I didn't mean to change what the current decorators do or how they should be used. I was only thinking about internal refactoring within the decorators. There could e.g. be a `_check_authenticated_and_service` helper function that is called from both decorator implementations. But the current code is just fine too.